distributed/shuffle/tests/test_shuffle.py

```diff
@@ -2281,3 +2282,77 @@ async def test_unpack_gets_rescheduled_from_non_participating_worker(c, s, a):
         shuffleA.block_barrier.set()
         result = await fut
         dd.assert_eq(result, expected)
+
+
+class FlakyConnectionPool(ConnectionPool):
```

Comment:
FYI this is now the third implementation of a  `FlakyConnectionPool`, all with slightly different implementations. It may be worth pulling that out but I'm fine with not doing it here.